apps/src/gamelab/GameLab.js

```diff
@@ -478,14 +413,127 @@ GameLab.prototype.execute = function() {
+      pwinMouseX: this.p5,
+      pwinMouseY: this.p5,
+      mouseButton: this.p5,
+      mouseIsPressed: this.p5,
+      touchX: this.p5,
+      touchY: this.p5,
+      ptouchX: this.p5,
+      ptouchY: this.p5,
+      touches: this.p5,
+      touchIsDown: this.p5,
+      pixels: this.p5,
+      deviceOrientation: this.p5,
+      accelerationX: this.p5,
+      accelerationY: this.p5,
+      accelerationZ: this.p5,
+      pAccelerationX: this.p5,
+      pAccelerationY: this.p5,
+      pAccelerationZ: this.p5,
+      rotationX: this.p5,
+      rotationY: this.p5,
+      rotationZ: this.p5,
+      pRotationX: this.p5,
+      pRotationY: this.p5,
+      pRotationZ: this.p5
+    }
+  });
+  this.JSInterpreter.onExecutionError.register(this.handleExecutionError.bind(this));
+  this.consoleLogger_.attachTo(this.JSInterpreter);
+  this.debugger_.attachTo(this.JSInterpreter);
+  this.JSInterpreter.parse({
+    code: this.studioApp_.getCode(),
+    blocks: dropletConfig.blocks,
+    blockFilter: this.level.executePaletteApisOnly && this.level.codeFunctions,
+    enableEvents: true
+  });
+  if (!this.JSInterpreter.initialized()) {
+    return;
+  }
+
+  this.p5specialFunctions.forEach(function (eventName) {
+    var func = this.JSInterpreter.findGlobalFunction(eventName);
+    if (func) {
+      this.eventHandlers[eventName] =
+          codegen.createNativeFunctionFromInterpreterFunction(func);
+    }
+  }, this);
+
+  codegen.customMarshalObjectList = [
+    window.p5,
+    window.Sprite,
+    window.Camera,
+    window.Animation,
+    window.p5.Vector,
+    window.p5.Color,
+    window.p5.Image,
+    window.p5.Renderer,
+    window.p5.Graphics,
+    window.p5.Font,
+    window.p5.Table,
+    window.p5.TableRow,
+    window.p5.Element
+  ];
+  // The p5play Group object should be custom marshalled, but its constructor
+  // actually creates a standard Array instance with a few additional methods
+  // added. The customMarshalModifiedObjectList allows us to set up additional
+  // object types to be custom marshalled by matching both the instance type
+  // and the presence of additional method name on the object.
+  codegen.customMarshalModifiedObjectList = [ { instance: Array, methodName: 'draw' } ];
+
+  // Insert everything on p5 and the Group constructor from p5play into the
+  // global namespace of the interpreter:
+  for (var prop in this.p5) {
+    this.JSInterpreter.createGlobalProperty(prop, this.p5[prop], this.p5);
+  }
+  this.JSInterpreter.createGlobalProperty('Group', window.Group);
+  // And also create a 'p5' object in the global namespace:
+  this.JSInterpreter.createGlobalProperty('p5', { Vector: window.p5.Vector });
+
+  /*
+  if (this.checkForEditCodePreExecutionFailure()) {
+   return this.onPuzzleComplete();
+  }
+  */
+};
+
 GameLab.prototype.onTick = function () {
   this.tickCount++;
 
   if (this.JSInterpreter) {
-    this.JSInterpreter.executeInterpreter(this.tickCount === 1);
+    this.JSInterpreter.executeInterpreter();
```

Comment:
We used to pass `true` as the first parameter when `tickCount === 1`, because that ensured that the first time we executed the interpreter, we would indicate that. Now, the first time we call `executeInterpreter` is inside the `preload` method, so there is no need to do that.
Upon reflection, it seems like the `JSInterpreter` object should know if that is the first call to `executeInterpreter` and the caller shouldn't need to do anything. But that will require a change across more apps..